### PR TITLE
Review memleaks fixes

### PR DESCRIPTION
**Pull request by azubieta**
The commits history is a little bit dirty. Do you want it this way or do you want me to CHANGE THE HISTORY :smile:

### CODE REVIEW

**Comment by TheAssassin** [comment]
@azubieta please implement the changes requested by codacy.

**Comment by TheAssassin** [comment]
@azubieta could you try to fix the remaining issues, please? (merge conflict and segfault of produced binaries)

**Comment by azubieta** [comment]
Sure!

**Comment by probonopd** [comment]
https://transfer.sh/Re7zm/appimaged-x86_64.AppImage (shared by @azubieta on IRC) does __not__ segfault for me.

**Comment by probonopd** [comment]
Seems to have reduced `RSS` too, but not much impact on `Kbytes`:

```
# The one before Azubieta

me@host:~$ appimaged --version
Version: continuous-2-g0d09f03

me@host:~$ sudo pmap -x $(pidof appimaged)
Address           Kbytes     RSS   Dirty Mode  Mapping
(...)
---------------- ------- ------- ------- 
total kB         12041860  112252   98764

me@host:~$ systemctl --user stop appimaged

# Now using
# https://transfer.sh/Re7zm/appimaged-x86_64.AppImage (shared by @azubieta on IRC) 

me@host:~$ Downloads/appimaged-x86_64.AppImage --appimage-version
Version: continuous-33-gb45d38a

me@host:~$ Downloads/appimaged-x86_64.AppImage

me@host:~$ sudo pmap -x $(pidof appimaged)
Address           Kbytes     RSS   Dirty Mode  Mapping
(...)
---------------- ------- ------- ------- 
total kB         12202096   29356   14068
```

Running `pmap -x` shows that apparently we are holding on to a lot of stuff (every processed AppImage) which we could let go.

**Comment by TheAssassin** [comment]
> https://transfer.sh/Re7zm/appimaged-x86_64.AppImage (shared by @azubieta on IRC) does not segfault for me. Yay! (Which commit was it built from?)

Why not ask it? We're hardcoding the version of appimagetool an AppImage has been built from into the runtime. Hence, calling the appimaged AppImage with `--appimage-version` will yield the commit hash.

**Comment by probonopd** [comment]
As written above ;-) it says:

```
# https://transfer.sh/Re7zm/appimaged-x86_64.AppImage (shared by @azubieta on IRC) 

me@host:~$ Downloads/appimaged-x86_64.AppImage --appimage-version
Version: continuous-33-gb45d38a
```

File: tests/test_libappimage.cpp [hunk 1]
```diff
@@ -163,9 +164,10 @@ TEST_F(AppImageTest, get_md5)
     std::string expected = "128e476a7794288cad0eb2542f7c995b";
     gchar * sum = get_md5("/tmp/testfile");
 
-    std::cout << sum;
+//    std::cout << sum;
```
Comment (TheAssassin) [inline tests/test_libappimage.cpp]: Could you please remove the comment entirely?

**Pull request merged**